Failing tests are passing locally and can't see how they could be related to the `HistoryCounter`:
- `FAILED lib/galaxy_test/selenium/test_history_multi_view.py::TestHistoryMultiView::test_list_list_display - selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".multi-history-panel .content-item[data-hid="5"][data-state="ok"]"}`
- `FAILED lib/galaxy_test/selenium/test_tool_form.py::TestLoggedInToolForm::test_run_apply_rules_tutorial - selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".multi-history-panel .content-item[data-hid="1"][data-state="ok"]"}`

--- a/client/src/components/History/CurrentHistory/HistoryCounter.vue
+++ b/client/src/components/History/CurrentHistory/HistoryCounter.vue
@@ -6,6 +6,7 @@
             variant="link"
             size="sm"
             class="rounded-0 text-decoration-none"
+            :disabled="!showControls"
             @click="onDashboard">
             <icon icon="database" />
             <span>{{ historySize | niceFileSize }}</span>
@@ -28,6 +29,7 @@
                 variant="link"
                 size="sm"
                 class="rounded-0 text-decoration-none"
+                :pressed="filterText == 'deleted:true'"
                 @click="toggleDeleted()">
                 <icon icon="trash" />
                 <span>{{ numItemsDeleted }}</span>
@@ -39,6 +41,7 @@
                 variant="link"
                 size="sm"
                 class="rounded-0 text-decoration-none"
+                :pressed="filterText == 'visible:false'"
                 @click="setFilter('visible:false')">
                 <icon icon="eye-slash" />
                 <span>{{ numItemsHidden }}</span>
@@ -73,6 +76,7 @@ export default {
         isWatching: { type: Boolean, default: false },
         lastChecked: { type: Date, default: null },
         filterText: { type: String, default: "" },
+        showControls: { type: Boolean, default: false },
     },
     data() {
         return {

--- a/client/src/components/History/CurrentHistory/HistoryPanel.vue
+++ b/client/src/components/History/CurrentHistory/HistoryPanel.vue
@@ -39,10 +39,10 @@
                         @update:history="$emit('updateHistory', $event)" />
                     <HistoryMessages :history="history" />
                     <HistoryCounter
-                        v-if="showControls"
                         :history="history"
                         :is-watching="isWatching"
                         :last-checked="lastChecked"
+                        :show-controls="showControls"
                         :filter-text.sync="filterText"
                         @reloadContents="reloadContents" />
                     <HistoryOperations

--- a/client/src/components/History/HistoryView.vue
+++ b/client/src/components/History/HistoryView.vue
@@ -1,6 +1,6 @@
 <template>
     <CurrentUser v-slot="{ user }">
-        <UserHistories v-if="user" v-slot="{ handlers }" :user="user">
+        <UserHistories v-if="user" v-slot="{ currentHistory, handlers }" :user="user">
             <div v-if="history" class="d-flex flex-column h-100">
                 <b-alert v-if="history.purged" variant="info" show>This history has been purged.</b-alert>
                 <div v-else class="flex-row flex-grow-0">
@@ -9,6 +9,7 @@
                         size="sm"
                         variant="outline-info"
                         title="Switch to this history"
+                        :disabled="currentHistory.id == history.id"
                         @click="handlers.setCurrentHistory(history)">
                         Switch to this history
                     </b-button>

